docs/targets/solana.rst

```diff
@@ -1,23 +1,52 @@
 Solana
 ======
 
-The Solana target requires `Solana <https://www.solana.com/>`_ v1.8.1.
-
-Solana has the following differences to Ethereum Solidity:
-
-- The address type is 32 bytes, not 20 bytes. This is what Solana calls an "account"
-- An address literal has to be specified using the ``address"36VtvSbE6jVGGQytYWSaDPG7uZphaxEjpJHUUpuUbq4D"`` syntax
-- There is no ``ecrecover()`` builtin function, but there is a ``signatureVerify()`` function which can check ed25519
-  signatures.
-- There is no concept of gas, so there is no gas functions
-- Solana balance is stored in a ``uint64``, so *address* ``.balance``, ``.transfer()`` and ``.send()``
-  all use ``uint64`` rather than ``uint256``.
-- Solana uses different accounts for the program code, and the contract data.
-- Programs are upgradable. There is no need to implement upgrades in Solidity.
+Solana Overview
+_______________
+
+The Solana blockchain has a unique account model that resembles a key-value database.
+As the underlying environment follows a programming model different than that of Ethereum, Solidity inner workings
+have been modified to work properly on Solana.
+
+A Solidity contract on Solana utilizes two accounts: a data account and a program account.
+The program account stores the contract's executable binary and owns the data account,
+which holds all the storage variables.
+
+Contract upgrades
++++++++++++++++++
+
+Provided that the data layout from a new contract is compatible with that of an old one, it is possible
+to update the binary in the program account and retain the same data, rendering
+contract upgrades implemented in Solidity unnecessary.
```

Comment:
This should say how to do those upgrades

Reply:
Added.

Reply:
This is very terse, it just deploys a program to known key pair (which we suggest anyway). Maybe a link to https://docs.solana.com/cli/deploy-a-program#set-a-programs-upgrade-authority would help.

Reply:
> This is very terse, it just deploys a program to known key pair (which we suggest anyway). Maybe a link to https://docs.solana.com/cli/deploy-a-program#set-a-programs-upgrade-authority would help.

The link you added here points to "Set an upgrade authority", but there is a section about redeploying a program (https://docs.solana.com/cli/deploy-a-program#redeploy-a-program). Which one should I add?

Reply:
I added a link to `Redeploy a program`. Let me know if this is what you were looking for.